lib/src/index.rs

```diff
@@ -67,6 +67,14 @@ pub trait Index: Send + Sync {
 
     fn common_ancestors(&self, set1: &[CommitId], set2: &[CommitId]) -> Vec<CommitId>;
 
+    /// Heads among all indexed commits.
+    ///
+    /// The index contains all historical heads and their ancestors and
```

Comment:
If I now understand what this does correctly, perhaps this should say "all the heads and their ancestors from the part of history that hasn't been garbage-collected"?

Reply:
It's kind of true if you think the index should include GC-ed commits, but the index (as for the default implementation) represents the corresponding operation `OP`, and the GC-ed (or abandoned to be more precise) operations are unreachable from the `OP`. So I think "all historical heads" is correct here.

Reply:
I see. I'll resolve this; I put a few more thoughts into https://github.com/martinvonz/jj/pull/2798#discussion_r1449361054. I also think the comments are good enough, this isn't blocking.